packages/traceability/issuer-irec-api/src/pods/certification-request/certification-request.dto.ts

```diff
@@ -0,0 +1,96 @@
+import { IntUnitsOfEnergy, PositiveBNStringValidator } from '@energyweb/origin-backend-utils';
+import { ApiProperty } from '@nestjs/swagger';
+import {
+    IsArray,
+    IsBoolean,
+    IsDate,
+    IsInt,
+    IsPositive,
+    IsString,
+    Min,
+    Validate,
+    ValidateIf
+} from 'class-validator';
+import { CertificationRequestStatus } from '@energyweb/issuer-api';
+
+export class CertificationRequestDTO {
+    @ApiProperty({ type: Number })
+    @IsInt()
+    @Min(0)
+    id: number;
+
+    @ApiProperty({ type: String })
+    @IsString()
+    deviceId: string;
+
+    @ApiProperty({ type: String })
+    @Validate(PositiveBNStringValidator)
+    @Validate(IntUnitsOfEnergy)
+    energy: string;
+
+    @ApiProperty({ type: String })
+    @IsString()
+    owner: string;
+
+    @ApiProperty({ type: Number })
+    @IsInt()
+    @IsPositive()
+    fromTime: number;
+
+    @ApiProperty({ type: Number })
+    @IsInt()
+    @IsPositive()
+    toTime: number;
+
+    @ApiProperty({ type: [String] })
+    @IsArray()
+    files: string[];
+
+    @ApiProperty({ type: Number })
+    @IsInt()
+    @IsPositive()
+    created: number;
+
+    @ApiProperty({ type: Boolean })
+    @IsBoolean()
+    approved: boolean;
+
+    @ApiProperty({ type: Boolean })
+    @IsBoolean()
+    revoked: boolean;
+
+    @ApiProperty({ type: Number })
+    @ValidateIf((dto: CertificationRequestDTO) => !!dto.requestId)
+    @IsInt()
+    @Min(0)
+    requestId?: number;
+
+    @ApiProperty({ type: Date, required: false })
+    @ValidateIf((dto: CertificationRequestDTO) => !!dto.approvedDate)
+    @IsDate()
+    approvedDate?: Date;
+
+    @ApiProperty({ type: Date, required: false })
+    @ValidateIf((dto: CertificationRequestDTO) => !!dto.revokedDate)
+    @IsDate()
+    revokedDate?: Date;
+
+    @ApiProperty({ type: Number, required: false })
+    @ValidateIf((dto: CertificationRequestDTO) => !!dto.issuedCertificateTokenId)
+    @IsInt()
+    @Min(0)
+    issuedCertificateTokenId?: number;
+
+    @ApiProperty({
+        enumName: 'CertificationRequestStatus',
+        enum: CertificationRequestStatus,
+        required: false
+    })
+    status?: CertificationRequestStatus;
+
+    @ApiProperty({ type: Boolean, required: false })
+    isPrivate?: boolean;
+
+    @ApiProperty({ type: String, required: false })
+    irecIssueId: string;
+}
```

Comment:
A lot of this code is copied from `@energyweb/issuer-api`.
As far as I can see, there is only one property added to the DTO.
Please use NestJS mapped types https://docs.nestjs.com/openapi/mapped-types#intersection

Something like:
```js
import { CertificationRequestDTO } from '@energyweb/issuer-api';

export class CertificationRequestIrec {
  @ApiProperty({ type: String, required: false })
    irecIssueId: string;
}

export class CertificationRequestIrecDTO extends IntersectionType(
  CertificationRequestDTO,
  CertificationRequestIrec,
) {}
```

And then use `CertificationRequestIrecDTO` in the controller.

Reply:
done